packages/effect/src/internal/stream.ts

```diff
@@ -7565,6 +7566,34 @@ export const zipLatest = dual<
   ): Stream.Stream<[A, A2], E2 | E, R2 | R> => pipe(self, zipLatestWith(that, (a, a2) => [a, a2]))
 )
 
+/** @internal */
+export type ZipLatestAllValues<T extends Array<Stream.Stream<unknown, unknown, unknown>>> = {
+  [K in keyof T]: T[K] extends Stream.Stream<infer U> ? U : never
+}
+/** @internal */
+export type ZipLatestAllErrors<T extends Array<Stream.Stream<unknown, unknown, unknown>>> = T extends
+  Array<Stream.Stream<unknown, infer E, unknown>> ? E : never
+/** @internal */
+export type ZipLatestAllContext<T extends Array<Stream.Stream<unknown, unknown, unknown>>> = T extends
+  Array<Stream.Stream<unknown, unknown, infer R>> ? R : never
+
+export const zipLatestAll = <T extends Array<Stream.Stream<unknown>>>(
+  ...streams: T
+): Stream.Stream<ZipLatestAllValues<T>, ZipLatestAllErrors<T>, ZipLatestAllContext<T>> => {
+  if (Arr.isTupleOf(streams, 1)) {
+    return pipe(streams[0], map((x) => [x] as ZipLatestAllValues<T>))
```

Comment:
BTW, @tim-smart , i also use `ZipLatestAllValues` type helper in 2 places inside the function (_here and 7 lines below from here_) for manual type-casting, because i can not think of a way to get rid of it. 
In total, `ZipLatestAllValues` is used in 4 places. Should i  still inline it?

Reply:
Yes still inline it :)